### ISSUE TITLE
Details in CNF blocks

### ISSUE THREAD

**Issue by gitlabspy** [RameenAbdal/StyleFlow#3]
Hi Rameen,

What an excellent work! I am trying to reimplement your work and I really want to know about the details in CNF block. Image below is from your paper, can you please tell me the `hidden units` and numbers of linear layers in each linear layer? It will help me a lot!
![cnf](https://user-images.githubusercontent.com/61307585/90468712-ac6e6680-e149-11ea-8cb9-7f2d8af1f08b.png)

Again, thx for your great work!!

**Comment by gitlabspy** [comment]
I use only one `Dense` layer for each `linear layer` in the figure given above and 512 hidden units each, after 40 epochs the results seem nice but not good enough I think. Could you tell me how many images/w s you used to train the flow please 😄 ?

**Comment by RameenAbdal** [comment]
Hi, 5k - 10k images/w-s should be fine. Make sure to use multiple CNF functions to train the flow network to avoid underfitting. For stable training, use moving batch norm layers. You can find the description in the Method section. Also, make sure to use edit specific subset layers to improve disentanglement.

**Comment by gitlabspy** [comment]
Thx! It does help a lot!!

**Comment by gitlabspy** [comment]
Oh, one more thing! `CNF functions` described in your paper means the neural net which is then solved by ode solver, right?
How many CNF blocks (in tensorflow probability they call it `ffjord` bijectors` you used in your implementation?

**Comment by RameenAbdal** [comment]
Yes, it's a neural net that is used by the ODE solver. One block worked for us.

**Comment by puppet101** [comment]
@gitlabspy 
Hi，could you please share your reimplement result? How is it compared with the authors?

**Comment by gitlabspy** [comment]
@puppet101 
Hi, my implementation result seems good but not as good as described in paper. I am pretty sure that it is just because of my implementation might not be correct or totally same as how the author did. 

In my implementation, the loss never reach -1000 not to mentioned the one posted in the paper😂 .. Here's a snippet of my implementation:
```
diffeq_layer = sequential([ 4* ConcatSquashLinear ]) # 4 stack of ConcatSquashLinear 
odefunc = ODEfunc(diffeq_layer) # wrapped with ODEFunc to calculate dx/dt
model = [CNF(odefunc) for i in num_blocks] # num_blocks=1 according to the paper
```
@RameenAbdal Is the way I implement correct?
